engine/derivative.cc

```diff
@@ -639,6 +640,31 @@ namespace MathDAG
        throw error("frac is not differentiable");
   }
   
+  template <>
+  NodePtr SystemOfEquations::derivative<>
+  (const OperationDAG<OperationType::percent>& expr)
+  {
+    if (expr.arguments[0].empty())
+      return zero;
+    else
+       throw error("percent is not differentiable");
+  }
+  
+//  template <>
+//  NodePtr SystemOfEquations::derivative
+//  (const OperationDAG<OperationType::fact>& expr)
+//  {
+//    if (expr.arguments[0].empty())
+//      return zero;
+//    else
+//      {
+//        //Expr x(expressionCache, expr.arguments[0][0]);
+//        //Expr gamma_p = boost::math::digamma(one+x)*boost::math::tgamma(one+x);
```

Comment:
gamma function is available in libmath - apparently, the one to use is tgamma(x) or tgammal(x) for the logarithm version.